Silent wdm logs

aleksandr-kotlyar opened the pull request:
Resolves #85 
## Use
To set silent WDM logs, set env variable in your project `os.environ['WDM_LOG_LEVEL']='0'`
To set another log levels, use numbers as their `logging.LEVEL`'s in standard library
```
    CRITICAL = 50
    ERROR = 40
    WARNING = 30
    INFO = 20
    DEBUG = 10
    NOTSET = 0
```
### Completed steps
- [x] `print` function changed to `logging` function
- [x] Pass env variable `WDM_LOG_LEVEL` as log level to `console()` logging function
- [x] Update README with new feature.
- [x] Fixed pack of README mismatches (Add Chromium example, move Edge example closer to IE, move github token information closer to configuration example and removed duplicated ff example, add honest webdriver import from selenium to all examples)

### Steps to Do
- [ ] load json to `logging.config.dictConfig(json)` from `logging_config.json`
- [ ] Write tests for `console()` log function and `WDM_LOG_LEVEL` env variable

pep8speaks commented:
Hello @aleksandr-kotlyar! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:









* In the file [`webdriver_manager/utils.py`](https://github.com/SergeyPirogov/webdriver_manager/blob/a15b687a68623333a7f529ff1bbcc96feebab092/webdriver_manager/utils.py):

> [Line 93:80](https://github.com/SergeyPirogov/webdriver_manager/blob/a15b687a68623333a7f529ff1bbcc96feebab092/webdriver_manager/utils.py#L93): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (111 > 79 characters)

SergeyPirogov commented:
Good, I like this approach

aleksandr-kotlyar commented:
@SergeyPirogov I request for help, i stuck with passing param to webdriver_manager library from external projects which use the lib. I am thinking about parsing command-line-arguments, but cannot find a good example or documentation.
I can try these options tomorrow but not sure it's the best or even working options. Please check the links.
1. https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument 
2. https://www.tutorialspoint.com/python/python_command_line_arguments.htm

SergeyPirogov commented:
Any params have to be passed as env variable

aleksandr-kotlyar commented:
@SergeyPirogov thanks for the tip. I almost finished. But need to add docs to README and tests for WDM_LOG_LEVEL env in `console()` method. Working on it now.

SergeyPirogov commented on webdriver_manager/utils.py:
```diff
@@ -84,6 +84,10 @@ def get_filename_from_response(response, name):
 
 
 def console(text: str, level=logging.INFO):
+    os_wdm_log_level = os.getenv('WDM_LOG_LEVEL')
```
Any default value here?

aleksandr-kotlyar replied:
Not yet. Does it need?

SergeyPirogov replied:
As I understand there has to be INFO level as default

SergeyPirogov replied:
Actually you have to pass it inside a method as level

aleksandr-kotlyar replied:
> As I understand there has to be INFO level as default

Yes, INFO is default.

> Actually you have to pass it inside a method as level

You mean this
```python 
def console(text: str, level=logging.INFO):
```
change to this?
```python
def console(text: str):
    level=logging.INFO
    os_wdm_log_level = os.getenv('WDM_LOG_LEVEL')
    if os_wdm_log_level:
        level = int(os_wdm_log_level)
```

SergeyPirogov replied:
```
def console(text: str):
    os_wdm_log_level = int(os.getenv('WDM_LOG_LEVEL', logging.INFO))
```

aleksandr-kotlyar replied:
Thank you. Now i see. I agree, it looks much better :)

aleksandr-kotlyar replied:
To be honest, i thought about separate `console()` log levels inside wdm, that's why i used `level` as `console()` argument. But ok, we can always return to this idea later.

aleksandr-kotlyar replied:
Done at https://github.com/SergeyPirogov/webdriver_manager/pull/106/commits/ff6695e64b165d5d0bf47b368655f5458a5fc85b

SergeyPirogov commented:
I see that log message looks like this:

```
 [WDM-INFO] Trying to download new driver from http://chromedriver.storage.googleapis.com/81.0.4044.138/chromedriver_linux64.zip (/home/travis/build/[secure]/webdriver_manager/webdriver_manager/utils.py:utils.py: 112)
```

Is is possible to delete from the output (/home/travis/build/[secure]/webdriver_manager/webdriver_manager/utils.py:utils.py: 112)

aleksandr-kotlyar commented:
> I see that log message looks like this:
> 
> ```
>  [WDM-INFO] Trying to download new driver from http://chromedriver.storage.googleapis.com/81.0.4044.138/chromedriver_linux64.zip (/home/travis/build/[secure]/webdriver_manager/webdriver_manager/utils.py:utils.py: 112)
> ```
> 
> Is is possible to delete from the output (/home/travis/build/[secure]/webdriver_manager/webdriver_manager/utils.py:utils.py: 112)

Sure! I thought it will be useful for some kind, of debugging problems, but for now it's useless while it doesn't show the row where console method is called.

SergeyPirogov commented:
Now it works as expected. The last question, what if we want to suppress all logs?

aleksandr-kotlyar commented:
>  what if we want to suppress all logs?
@SergeyPirogov give me more details, please. :) What do u mean "suppress all logs"?
- all webdriver_manager custom "console()" function logs
- all pytest logs in webdriver_manager's TravisCI
- all logs from the whole entire world? ;)

Give me the case: What do you want to do, and what do you expect to get?

SergeyPirogov commented:
Now logs looks like this:
```
tests/test_chrome_driver.py::test_driver_can_be_saved_to_custom_path 2020-05-12 07:45:49,050 [WDM-INFO] Looking for [chromedriver 2.26 linux64] driver in cache 
2020-05-12 07:45:49,050 [WDM-INFO] File found in cache by path [/home/travis/build/[secure]/webdriver_manager/custom/drivers/chromedriver/2.26/linux64/chromedriver]
```

What if I want to hide logs from WDM? What I have to do? just pul log level noset or smt else?

aleksandr-kotlyar commented:
> What if I want to hide logs from WDM? What I have to do? just pul log level noset or smt else?
Yes, just pul log level = '0'.
```python
os.environ['WDM_LOG_LEVEL']='0'
```

SergeyPirogov commented:
Could you please add this to Readme and I will merge?

SergeyPirogov commented:
I think we can make it even better, let's create our own class Logger that will be a wrapper for a standard logger with a single method log("text here"). This log method implementation is the same as a console, but in such case, we will be able to pass the logging level as a parameter:

Example:

```
from webdriver_manager.firefox import GeckoDriverManager
GeckoDriverManager(logLevel=logging.INFO).install()

```

aleksandr-kotlyar commented:
> I think we can make it even better, let's create our own class Logger that will be a wrapper for a standard logger with a single method log("text here"). This log method implementation is the same as a console, but in such case, we will be able to pass the logging level as a parameter:
> 
> Example:
> 
> ```
> from webdriver_manager.firefox import GeckoDriverManager
> GeckoDriverManager(logLevel=logging.INFO).install()
> ```
> 
> In such a case, we will have the ability to set log level either using the env variable or constructor parameter

@SergeyPirogov Understood. Will do it in the next day.

SergeyPirogov commented:
@aleksandr-kotlyar Do we have a chance to finish it this week?

aleksandr-kotlyar commented:
> Do we have a chance to finish it this week?

@SergeyPirogov Sure. Documentation and tests will be ready this week. I will try to finish today evening, so you could merge it.

aleksandr-kotlyar commented:
But refactoring this way - i think it will be better todo in next iteration and PR.
```
from webdriver_manager.firefox import GeckoDriverManager
GeckoDriverManager(logLevel=logging.INFO).install()
```
I really don't understand right now why you think it's better than set env in the project.
With env + this way you will have two places where you can set this. And it's much harder to support. So i think we need more details how you wish to see this option. Cause passing many arguments through all classes - doesn't sound good. And defining class WdmLogger inside test - it's more tough than define env var.

aleksandr-kotlyar commented:
@SergeyPirogov i have finished the work.
Updated Readme with the feature https://github.com/SergeyPirogov/webdriver_manager/pull/106/commits/e8bbc2f6c6daa881fe8af0bee072f2ba192d5ff1
Fixed pack of README mismatches https://github.com/SergeyPirogov/webdriver_manager/pull/106/commits/b0e67648dbe789a45760aa4e22f7e491261ba1dd

Also tried to complete these steps, but gave up after 4 hours.
> ### Steps to Do
>  - [ ] load json to logging.config.dictConfig(json) from logging_config.json
>  - [ ] Write tests for console() log function and WDM_LOG_LEVEL env variable

SergeyPirogov commented:
What about passing logLevel via construsto? I described this above. To create logger class and pass logging level as ChromeDriverManager(logLevel=0)

SergeyPirogov commented:
Thank you!

aleksandr-kotlyar commented:
> What about passing logLevel via construsto? I described this above. To create logger class and pass logging level as ChromeDriverManager(logLevel=0)

I recommend to do it in next PR. I have many questions: https://github.com/SergeyPirogov/webdriver_manager/pull/106#issuecomment-627867024
